pkg/bpf/collection.go

```diff
@@ -41,6 +48,135 @@ func LoadCollectionSpec(path string) (*ebpf.CollectionSpec, error) {
 	return spec, nil
 }
 
+func RemoveUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	return removeUnreachableTailcalls(spec)
+}
+
+func removeUnreachableTailcalls(spec *ebpf.CollectionSpec) error {
+	type TailCall struct {
+		referenced bool
+		visited    bool
+		spec       *ebpf.ProgramSpec
+	}
+
+	entrypoints := make([]*ebpf.ProgramSpec, 0)
+	tailcalls := make(map[uint32]*TailCall)
+
+	for _, prog := range spec.Programs {
+		var id, slot uint32
+		// Consider any program that doesn't follow the tailcall naming convention
+		// x/y to be an entrypoint.
+		if _, err := fmt.Sscanf(prog.SectionName, "%d/%v", &id, &slot); err != nil {
+			entrypoints = append(entrypoints, prog)
+			continue
+		}
+
+		const (
+			cilium_calls_map = 2
```

Comment:
How is this magic number going to stay in sync?

Reply:
We can consider removing the map ID system by using section names like `cilium_calls/23`, but that would require some work on the ELF rewriter since they're actually called `cilium_calls_01234` in the ELF.

If/when we remove ELF rewriting, my intention was to fix the map name to `cilium_calls` in the ELF and change the `MapSpec.Name` at runtime instead. This results in `CollectionSpec.Maps["cilium_calls"]` but `MapSpec.Name = "cilium_calls_01234"`, so the map can always be retrieved using a stable key.

Reply:
We had a meeting since, agreed that removing the ID system is the goal long term, but since that will take a bit of time to move forward with the magic IDs for now.